src/System.Text.Json.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -333,9 +333,9 @@ public bool Equals(JsonProperty other)
             public override int GetHashCode()
             {
                 int result = Object.GetHashCode();
-                result = result * 19 + (byte)_name[0];
-                result = result * 19 + (byte)_name[_name.Length - 1];
-                result = result * 19 + (byte)_name[_name.Length>>2];
+                result = result * 19 + (byte)_name.Bytes[0];
```

Comment:
In general, you should get Bytes once, store it in a local and then use it (if you need to use it more than once).